proxy/src/console/provider/neon.rs

```diff
@@ -277,6 +281,14 @@ impl super::Api for Api {
             return Ok(cached);
         }
 
+        // check rate limit
+        if !self
+            .endpoint_rate_limiter
+            .check(user_info.endpoint.normalize().into(), 1)
```

Comment:
Also, another question is if we need to normalize it here or not. Technically, pgbouncer has a different address, so, maybe it's not necesary, but I am not sure. So, it's up to you.